core/main/src/firebolt/firebolt_gateway.rs

```diff
@@ -312,33 +296,55 @@ impl FireboltGateway {
     }
 }
 
-fn validate_request(open_rpc_state: OpenRpcState, request: &RpcRequest) -> Result<(), String> {
-    let major_version = open_rpc_state.get_version().major.to_string();
-    let openrpc_validator = open_rpc_state.get_openrpc_validator();
-
-    if let Some(rpc_method) = openrpc_validator.get_method_by_name(&request.method) {
-        let validator = openrpc_validator
-            .params_validator(major_version, &rpc_method.name)
-            .unwrap();
-
-        if let Ok(params) = serde_json::from_str::<Vec<serde_json::Value>>(&request.params_json) {
-            if params.len() > 1 {
-                if let Err(errors) = validator.validate(&params[1]) {
-                    let mut error_string = String::new();
-                    for error in errors {
-                        error_string.push_str(&format!("{} ", error));
-                    }
-                    return Err(error_string);
+fn validate_request(open_rpc_state: OpenRpcState, request: &mut RpcRequest) -> Result<(), String> {
+    // Params should be valid given we get the request from Firebolt WS Call context is decorated
+    // in index 0
+    if let Ok(params) = serde_json::from_str::<Vec<serde_json::Value>>(&request.params_json) {
+        // Check if there are params
+        let param = params.get(1);
+        if param.is_none() {
+            // if params are necessary then handler or jq rule will fail
+            // This method can unblock other calls.
+            return Ok(());
+        }
+        let param = param.unwrap();
```

Comment:
unsafe unwrap()

Reply:
param is checked in line number 305 so its not unsafe